packages/teleport/src/DesktopSession/useDesktopSession.tsx

```diff
@@ -47,6 +47,8 @@ export default function useDesktopSession() {
 
   const [canShareDirectory, setCanShareDirectory] = useState(false);
   const [isSharingDirectory, setIsSharingDirectory] = useState(false);
+  const [directorySharingBrowserErr, setDirectorySharingBrowserErr] =
```

Comment:
Instead of 3 booleans, would it make more sense to have a directory sharing state? It could encode all of this info:

- has permissions to share
- is actively sharing
- failed to share due to an error

Reply:
59af539a62d4d09ccc2f16b8280f185d9378d786